awx/api/views/__init__.py

```diff
@@ -3824,6 +3824,82 @@ def get_queryset(self):
         return job.get_event_queryset().select_related('host').order_by('start_line')
 
 
+class JobJobEventsChildrenSummary(APIView):
+    '''
+    Special view to facilitate processing job output in the UI.
+    In order to collapse events and their children, the UI needs to know how
+    many children exist for a given event.
+    This view returns a JSON object where the key is the event counter, and the value
+    is the number of children (and grandchildren) events.
+    Only events with children are included in the output.
+
+    e.g. Demo Job Template job
+    given:
```

Comment:
By "given", you just mean that this is the event data for the Demo Job Template?

Reply:
exactly